### ISSUE TITLE
Redesign and drastically simplify tx fee/prioritization policy

### ISSUE THREAD

**Issue by daira** [zcash/zcash#1896]
Bitcoin fee/prioritisation policy was based on the assumption of the monetary amounts transferred being visible. This assumption no longer holds. IMHO there is no reason for these amounts to be an input to tx prioritisation at all. We should be trying to minimise tx acceptance latency; the simplest way to do that is for priority to be based mainly on transaction age, with as little complication as possible introduced to mitigate denial of service attacks.

**Comment by nathan-at-least** [comment]
I strongly disagree. The txn fee is the primary way a user can express how important quick mining is to them.

I _might_ be convinced for some combination of metrics like fee & age, but I will also heavily favor a policy that arguably maximizes miner revenue. (I saw arguably because there may be alternatives and no way for us to determine which would actually maximize revenue, but I at least want a compelling argument.)

**Comment by daira** [comment]
What precisely do you disagree with:
1. that there is no need for the monetary amounts other than the fee to be an input to tx prioritisation;
2. that we should be trying to minimise tx acceptance latency;
3. that the simplest way to do that is for priority to be based mainly on transaction age, with as little complication as possible introduced to mitigate denial of service attacks

?

**Comment by nathan-at-least** [comment]
Ah, I misunderstood your proposal thinking you meant prioritization should be insensitive entirely to fees, but I misread. You mean all of the non-fee value, correct? In that case I agree with all three points in your last comment.

**Comment by daira** [comment]
See also #2133.

**Comment by nathan-at-least** [comment]
Let's focus on this in an upcoming topical meeting.

**Comment by str4d** [comment]
Upstream have removed coin age priority and free transactions for 0.15:

- Non-technical discussion: https://github.com/bitcoin/bitcoin/issues/9601
- Removal PR: https://github.com/bitcoin/bitcoin/pull/9602

**Comment by daira** [comment]
See also previous upstream discussion at https://github.com/bitcoin/bitcoin/issues/6675 .

**Comment by daira** [comment]
I want us to be very careful to understand what the upstream PR is doing before we decide to backport it. Bitcoin Core's policy goals for transaction fees seem to be quite different from ours; at the risk of oversimplifying, we want fees to be as low as they possibly can be subject to mitigating denial of service (and our blockchain is at a different stage of development to Bitcoin both wrt DoS and the relative importance of block rewards vs fees).

**Comment by daira** [comment]
I've briefly looked at the upstream PR; overall it looks to be going in the wrong policy direction for us, and in any case it would be too much code to review for 1.0.9.

**Comment by daira** [comment]
I notice that we switched the default for `-relaypriority` to off in #1001, motivated by allowing spending of mining outputs during slow start. It's still defaulted to off, unlike in Bitcoin: https://github.com/zcash/zcash/blob/master/src/main.cpp#L1540

**Comment by daira** [comment]
We plan to backport the removal of priority and "free transactions" as part of the ZIP 317-related changes.
